// @uifabric/example-app-base/lib-amd/utilities/createApp.js
var __assign = (this && this.__assign) || Object.assign || function(t) {
    for (var s, i = 1, n = arguments.length; i < n; i++) {
        s = arguments[i];
        for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
            t[p] = s[p];
    }
    return t;
};
define(["require", "exports", "react", "react-dom", "../components/App/App", "office-ui-fabric-react/lib/utilities/router/index", "office-ui-fabric-react/lib/Utilities", "office-ui-fabric-react/lib/Fabric"], function (require, exports, React, ReactDOM, App_1, index_1, Utilities_1, Fabric_1) {
    "use strict";
    Object.defineProperty(exports, "__esModule", { value: true });
    function createApp(examples, hideChrome) {
        var rootElement;
        var groups = !Array.isArray(examples) ? [examples] : examples;
        function _onLoad() {
            rootElement = document.createElement('div');
            document.body.appendChild(rootElement);
            Utilities_1.setBaseUrl('./dist/');
            var routes = groups.map(function (group, groupIndex) { return group.examples.map(function (example, index) { return (React.createElement(index_1.Route, { key: example.key, path: '#component=' + example.key, component: example.onRender })); }); });
            ReactDOM.render(React.createElement(Fabric_1.Fabric, null,
                React.createElement(index_1.Router, null,
                    React.createElement(index_1.Route, { key: 'minimal', path: '?minimal', component: function (props) { return React.createElement("div", __assign({}, props)); } }, routes),
                    React.createElement(index_1.Route, { key: 'app', component: function (props) { return React.createElement(App_1.App, __assign({ appDefinition: _getDefinition(groups) }, props)); } }, routes))), rootElement);
        }
        function _onUnload() {
            if (rootElement) {
                ReactDOM.unmountComponentAtNode(rootElement);
                rootElement = null;
            }
        }
        var isReady = document.readyState === 'interactive' || document.readyState === 'complete';
        if (isReady) {
            _onLoad();
        }
        else {
            window.onload = _onLoad;
        }
        window.onunload = _onUnload;
    }
    exports.createApp = createApp;
    function _getDefinition(groups) {
        return {
            appTitle: 'Fabric Examples',
            testPages: [],
            examplePages: groups.map(function (group, groupIndex) { return ({
                name: group.title,
                links: group.examples.map(function (example, exampleIndex) { return ({
                    component: example.onRender,
                    key: example.key,
                    name: example.title,
                    url: '#component=' + example.key
                }); })
            }); }),
            headerLinks: [
                {
                    name: 'Getting started',
                    url: '#/'
                },
                {
                    name: 'Fabric',
                    url: 'http://dev.office.com/fabric'
                },
                {
                    name: 'Github',
                    url: 'http://www.github.com/officedev'
                }
            ]
        };
    }
});
